src/prefect/core/flow.py

```diff
@@ -994,6 +994,9 @@ def _run(
             "context", {}
         ).copy()  # copy to avoid modification
 
+        # set flow_run_id from args or uuid if flow_run_id is not an argument
+        flow_run_context.setdefault("flow_run_id", kwargs.pop("flow_run_id", {}))
```

Comment:
```suggestion
        flow_run_context.setdefault(
            "flow_run_id", kwargs.pop("flow_run_id", str(uuid.uuid4()))
        )
```

One last thing - your implementation pops `flow_run_id` from kwargs but, if not found, defaults to an empty dictionary.  Instead we want the default to be a uuid - this suggestion fixes that 👍